mailet/amqp/src/main/java/org/apache/james/transport/mailets/AmqpForwardAttribute.java

```diff
@@ -183,6 +195,12 @@ String preInit(MailetConfig mailetConfig) throws MailetException {
             throw new MailetException("No value for " + EXCHANGE_PARAMETER_NAME
                     + " parameter was provided.");
         }
+        exchangeType = mailetConfig.getInitParameter(EXCHANGE_TYPE_PARAMETER_NAME);
+        if (StringUtils.isNotEmpty(exchangeType) && !VALIDATE_EXCHANGE_TYPES.contains(exchangeType)) {
```

Comment:
This condition will evaluate to `false` if `exchangeType` is empty and thus the exception won't be thrown, empty doesn't seem like a valid value here. Maybe change the && to ||?

If this was done on purpose to use the default exchange type when empty, I think this should be specified, preferably using a unit test.

Reply:
the empty `exchangeType` (from mailet parameter configuration) is accepted. 
it help compatibility with old code. 

The test case already exists in AmqpForwardAttributeTest.java (the old code passes).